llvmlite/ir/module.py

```diff
@@ -31,6 +31,23 @@ def add_metadata(self, operands):
             md = self._metadatacache[key]
         return md
 
+    def add_debug_info(self, kind, operands, is_distinct=False):
+        """
+        Add debug information metadata to the module with the given
+        *operands* (a dict of values with string keys) or return
+        a previous equivalent metadata. A DIValue instance is returned,
+        it can then be associated to e.g. an instruction.
+        """
+        n = len(self.metadata)
+        operands = tuple((key, operands[key]) for key in operands)
```

Comment:
Since a dict in unordered, you may want to sort the operands if you want to make the cache more efficient (e.g. `operands = tuple(sorted(operands.items()))`.